get-resource.sh

```diff
@@ -21,13 +21,28 @@ curl_with_flags() {
     curl "$@"
 }
 
+get_filename_without_extension() {
+	local filename=$1
+	# tar supported extensions are listed here:
+	# https://www.gnu.org/software/tar/manual/html_section/Compression.html#auto_002dcompress
+	declare -a extensions=(gz tgz taz Z taZ bz2 tz2 tbz2 tbz lz lzma tlz lzo xz zst tzst)
```

Comment:
Some of these extensions are also including the `.tar` part, for example `.tgz` is short for `.tar.gz`, but in code we check always for `.tar.${ex}` which makes it incorrect.

I think we should rewrite this list to have full extension pairs, ie. `extensions=(tar.gz tgz ...)`

Reply:
That is a good point as well, shortens the list quite a bit for maintainability.